daphne/src/roles/leader.rs

```diff
@@ -22,47 +22,54 @@ use crate::{
     DapResource, DapResponse, DapTaskConfig, DapVersion, MetaAggregationJobId,
 };
 
-macro_rules! leader_post {
-    (
-        $role:expr,
-        $task_id:expr,
-        $task_config:expr,
-        $path:expr,
-        $req_media_type:expr,
-        $resp_media_type:expr,
-        $resource:expr,
-        $req_data:expr,
-        $is_put:expr
-    ) => {{
-        let url = $task_config
-            .helper_url
-            .join($path)
-            .map_err(|e| fatal_error!(err = e))?;
-
-        let req = DapRequest {
-            version: $task_config.version,
-            media_type: $req_media_type,
-            task_id: Some($task_id.clone()),
-            resource: $resource,
-            payload: $req_data,
-            url,
-            sender_auth: Some(
-                $role
-                    .authorize(&$task_id, &$req_media_type, &$req_data)
-                    .await?,
-            ),
-            taskprov: None,
-        };
+struct LeaderPostOptions<'p> {
+    path: &'p str,
+    req_media_type: DapMediaType,
+    resp_media_type: DapMediaType,
+    resource: DapResource,
+    req_data: Vec<u8>,
+    is_put: bool,
```

Comment:
If set, this PUTs instead of POSTs. Since we're changing this code, we might as well rename it to be more accurate. How about "leader_send_http_request" and "LeaderHttpRequestOptions"?